service-deployment/src/end-to-end-test/etc/launcher.json

```diff
@@ -15,7 +15,16 @@
         "resource": {
           "cpuNumber": 1,
           "memoryMB": 512,
-          "portRanges": [],
+          "portDefinitions": {
+            "http": {
+              "start": 0,
+              "count": 1
```

Comment:
If we don't care the port numbers just count, can we just omit this field?

Reply:
"start": 0 means any port.  You mean only need set count and ignore start ?  

the current code is not support this scenario, the framework will failed at Json mapping to class if "start" is not set. 

I am not sure if the reset-server can auto-fill the "start" if user not set it in user json. 

 

---
In reply to: [181942244](https://github.com/Microsoft/pai/pull/478#discussion_r181942244) [](ancestors = 181942244)

Reply:
This test json is sent to framework launcher directly.